src/sqlfluff/core/rules/base.py

```diff
@@ -237,18 +250,21 @@ def is_just_source_edit(self) -> bool:
             and self.edit[0].raw == self.anchor.raw
         )
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         if self.edit_type == "delete":
             detail = f"delete:{self.anchor.raw!r}"
         elif self.edit_type in ("replace", "create_before", "create_after"):
             if hasattr(self.edit, "raw"):
-                new_detail = self.edit.raw  # pragma: no cover TODO?
+                seg = cast(BaseSegment, self.edit)  # pragma: no cover
+                new_detail = seg.raw  # pragma: no cover TODO?
             else:
```

Comment:
Now that we've got stronger typing in place, I think that we can remove this first clause. `edit` is defined as `Optional[Iterable[BaseSegment]]` so it should never have a `raw` attribute (which is from the days when we accepted a single segment).

(We also coerce it to a `list` in the `__init__` method)